src/go/rpk/pkg/api/admin/admin.go

```diff
@@ -153,6 +193,28 @@ func (a *AdminAPI) sendAny(method, path string, body, into interface{}) error {
 	return maybeUnmarshalRespInto(method, url, res, into)
 }
 
+// sendToLeader sends a single request to the leader of the Admin API for Redpanda >= 21.11.1
+// otherwise, it broadcasts the request
+func (a *AdminAPI) sendToLeader(method, path string, body, into interface{}) error {
+	if a.brokerIdToUrls == nil {
```

Comment:
>The logic should probably be to populate brokerIdToUrls partially if any nodes fail to respond, and then in this function only fall back to sendAll if brokerIdToUrls does not contain the ID of the leader.

👍 

>I'd also suggest to populating that mapping it within sendToLeader rather than in the class constructor

In order to do the mapping of brokerID to brokerURL we need to send a request to each broker.

If we do that every time we try to make a write, 🤔 wouldn't that defeat the purpose of this PR as we'll be always doing N requests (N being the number of brokers)?

>that way we know we're dealing with reasonably fresh data, and not going to deal with weirdness if a node was offline at construction time but later came online

Is it possible that brokers change broker ID?